src/aosm/azext_aosm/_configuration.py

```diff
@@ -137,22 +124,91 @@ def __post_init__(self):
         if isinstance(self.arm_template, dict):
             self.arm_template = ArtifactConfig(**self.arm_template)
 
+    def validate(self):
+        ## validate that all of the configuration parameters are set
+
+        if self.location == DESCRIPTION_MAP["location"] or "":
+            raise ValueError("Location must be set")
+        if self.publisher_name == DESCRIPTION_MAP["publisher_name_nsd"] or "":
+            raise ValueError("Publisher name must be set")
+        if (
+            self.publisher_resource_group_name
+            == DESCRIPTION_MAP["publisher_resource_group_name_nsd"]
+            or ""
+        ):
+            raise ValueError("Publisher resource group name must be set")
+        if (
+            self.acr_artifact_store_name == DESCRIPTION_MAP["acr_artifact_store_name"]
+            or ""
+        ):
+            raise ValueError("ACR Artifact Store name must be set")
+        if (
+            self.network_function_definition_group_name
+            == DESCRIPTION_MAP["network_function_definition_group_name"]
+            or ""
+        ):
+            raise ValueError("Network Function Definition Group name must be set")
+        if (
+            self.network_function_definition_version_name
+            == DESCRIPTION_MAP["network_function_definition_version_name"]
+            or ""
+        ):
+            raise ValueError("Network Function Definition Version name must be set")
+        if (
+            self.network_function_definition_offering_location
+            == DESCRIPTION_MAP["network_function_definition_offering_location"]
+            or ""
+        ):
+            raise ValueError(
+                "Network Function Definition Offering Location must be set"
+            )
+        if self.nsdg_name == DESCRIPTION_MAP["nsdg_name"] or "":
+            raise ValueError("NSDG name must be set")
+        if self.nsd_version == DESCRIPTION_MAP["nsd_version"] or "":
+            raise ValueError("NSD Version must be set")
+
     @property
     def build_output_folder_name(self) -> str:
         """Return the local folder for generating the bicep template to."""
-        folder_path = self.bicep_output_folder_path
-        return f"{folder_path}/{NSD_DEFINITION_OUTPUT_BICEP_PREFIX}"
+        current_working_directory = os.getcwd()
```

Comment:
I don't think you need to do this. I think you can just do os.mkdir and it will make you a folder in the current directory

Reply:
I think I still want to have this because I need a reference to the path when I am creating sub directories